### PR TITLE
move uint64s to beginning of struct to ensure they are 64-bit aligned

### PR DESCRIPTION
**Pull request by wnienhaus**
## Description
This fixes an "unaligned 64-bit atomic operation" panic on 32-bit ARM, (in my case a Raspberry Pi 4, running 32-bit Raspbian).

The panic happens (at least) during the first login after a service (re)start.

This issue has been spotted in other Golang projects too. (e.g. https://github.com/census-instrumentation/opencensus-go/commit/01b39586f932bbb0b4e1cb6499e454c470e9138b)

The Golang project documents this at https://golang.org/pkg/sync/atomic/#pkg-note-BUG mentioning it is the "caller's responsibility to arrange for 64-bit alignment of 64-bit words accessed atomically" and "The first word in (...) an allocated struct (...) can be relied upon to be 64-bit aligned"

## Related Issue
- Fixes issue #1887 

## Motivation and Context
It fixes a crash on 32-bit ARM (and likely other 32-bit systems).

## How Has This Been Tested?
- Repeated the following multiple times without the fix: Restart ocis server (kubernetes pod) and attempt to login with a web browser with "einstein:relativity". Panic was repeatable every time.
- Then build a new docker image with the fix
- And again repeated multiple times: Restart ocis server and attempt login as einstein. Panic no longer repeatable.
- (Note: the Kubernetes pod was technically not "restarted" but essentially recreated each time because there is no restart in Kubernetes. However, because /var/tmp/ocis were mounted-in as a volume and server.{key,crt} and ldap.(key,crt} were also supplied from the outside in (secrets mounted into the pod), this should in effect amount to a restart).

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:
- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Review by fschade — APPROVED** [review]
Thanks for taking care, LGMT 👍

File: ocis-pkg/sync/cache.go [hunk 1]
```diff
@@ -8,10 +8,10 @@ import (
 
 // Cache is a barebones cache implementation.
 type Cache struct {
-	entries  sync.Map
-	pool     sync.Pool
 	capacity uint64
 	length   uint64
```
Comment (fschade) [inline ocis-pkg/sync/cache.go:12]: // has to be the first words
// in order to be 64-aligned on 32-bit architectures.
capacity, length int64 // access atomically

**Review by fschade — CHANGES_REQUESTED** [review]
Maybe it's also good to add a comment for later reference.

File: ocis-pkg/sync/cache.go (hunk 1 above)
Comment (wnienhaus) [reply ocis-pkg/sync/cache.go]: Good point. Added.

**Comment by wnienhaus** [comment]
> Thanks for taking care, @wnienhaus can you please add a changelog and comments.
> Then LGMT +1

Both are added now.

**Review by kulmann — APPROVED** [review]
Thanks for your contribution! LGTM 👍

File: ocis-pkg/sync/cache.go (hunk 1 above)
Comment (fschade) [reply ocis-pkg/sync/cache.go]: Thanks 🤩

**Review by fschade — APPROVED** [review]
LGMT 👍

**Review by phil-davis — APPROVED** [review]

**Comment by skygunner** [comment]
also happens to 64bit arm

**Pull request merged**